providers/google.go

```diff
@@ -133,25 +181,20 @@ func (g *GoogleOp) requestTokens(ctx context.Context, cicHash string) ([]byte, e
 		}
 	}
 
-	http.Handle(g.CallbackPath, rp.CodeExchangeHandler(marshalToken, provider))
-
-	lis := fmt.Sprintf("localhost:%s", g.RedirURIPort)
-	g.server = &http.Server{
-		Addr: lis,
-	}
-
-	logrus.Infof("listening on http://%s/", lis)
-	logrus.Info("press ctrl+c to stop")
-	earl := fmt.Sprintf("http://localhost:%s/login", g.RedirURIPort)
-	util.OpenUrl(earl)
+	callbackPath := redirectURI.Path
+	http.Handle(callbackPath, rp.CodeExchangeHandler(marshalToken, provider))
 
 	go func() {
-		err := g.server.ListenAndServe()
+		err := g.server.Serve(ln)
 		if err != nil && err != http.ErrServerClosed {
 			logrus.Error(err)
 		}
 	}()
 
+	loginURI := fmt.Sprintf("http://localhost:%s/login", redirectURI.Port())
+	logrus.Infof("Opening browser to on http://%s/", loginURI)
+	util.OpenUrl(loginURI)
```

Comment:
I might make a PR after this one is merged to add an option in `provider.Options` to disable automatically opening the browser (useful for integration tests). After that, the only difference in `opk-ssh`'s `GoogleOp` is refresh functionality I believe, so perhaps we can get closer to reusing the one here in `openpubkey` without having to duplicate the same core logic.

Reply:
Created an issue for this
https://github.com/openpubkey/openpubkey/issues/169